pkg/providers/cloudstack/cloudstack.go

```diff
@@ -40,6 +40,8 @@ import (
 const (
 	eksaLicense                = "EKSA_LICENSE"
 	controlEndpointDefaultPort = "6443"
+	nodeFailuredomainLabel     = "cluster.x-k8s.io/failure-domain"
+	nodeFailuredomainLabelValue = "'{{ds.meta_data.availability_zone}}'"
```

Comment:
where does `availability_zone` come from on the host metadata?

Reply:
This is a built-in meta_data provided by cloudstack platform. Cloud-init jinja's template engine has the value to replace it with machine's zone name.

Reply:
One concern is that the value of `ds.meta_data.availability_zone` is controlled by cloudstack platform and currently its value is the zone name where node/machine is deployed.

This availability_zone name maybe different from a higher level abstract failure_domain name customer defined in cluster.yaml file.

for example: 
`apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: CloudStackDatacenterConfig
metadata:
  name: wanyufe
spec:
  availabilityZones:
  - name: az1
    domain: ROOT
    account: admin
    zone:
      name: zone1
      network:
        name: GuestNet1
    credentialsRef: global
    managementApiEndpoint: http://192.168.1.130:8080/client/api
---
`

the `ds.meta_data.availability_zone` value is zone1, but customer may want to see az1 as failure_domain. 

If so, we can provide a different name like `ds.meta_data.failure_domain`, and replace it with failure domain name in CAPC code. This needs to be discussed.

Reply:
Agreed that Zone alone is no longer accurate.  Let's open an issue for this.